modules/drivers/canbus/can_client/socket/socket_can_client_raw.cc

```diff
@@ -71,22 +69,23 @@ ErrorCode SocketCanClientRaw::Start() {
     return ErrorCode::CAN_CLIENT_ERROR_BASE;
   }
 
-  // init config and state
-  // 1. set receive message_id filter, ie white list
+  // Init config and state
+  // 1. Set receive message_id filter, i.e, the white list
   struct can_filter filter[2048];
-  for (int i = 0; i < 2048; ++i) {
+  for (int i = 0; i < sizeof(filter); ++i) {
```

Comment:
Probably a bug: `sizeof(filter)` will be `2048 * sizeof(can_filter)`.  Please change back to 2048

Reply:
The size of the array can be calculated with `sizeof(filter)/sizeof(filter[0])` , which will evaluate at compile time(None VLA) .